dev/packages/alpha/netflow-0.0.2/dataset/log/fields/ecs.yml

```diff
@@ -0,0 +1,3141 @@
+- name: '@timestamp'
```

Comment:
Does this only contain the fields that are used by this dataset or is this all of ECS? The goal should be to only have the fields in here that are actually used.

Reply:
So using the netflow module as an example the fields consist of:

- some base filebeat fields that may not be in ECS (like input.type)
- all filebeat netflow input fields (some 400+ fields)
- a subset of ECS namespaces (like `agent`, `source`, `destination`, `network`, `event`)

Very few of these fields are something that the module actually creates. This module basically brings together Filebeat with an Ingest Node pipeline to add geo location and autonomous system numbers. So keeping this module updated will be a lot of work. And since it depends on Filebeat's fields, this template will vary based on the Filebeat version.

So I wonder if it would be possible to list some dependent templates in addition to custom fields created by a module to create a composite. Like saying that for this module it depends on

- filebeat $automatic_version | include_namespaces(agent | netflow | input.type | source | destination)
- ECS 1.4 | include_namespaces(source | destination)

<img width="695" alt="Screen Shot 2020-04-28 at 10 35 55 AM" src="https://user-images.githubusercontent.com/4565752/80500474-41034d80-893c-11ea-97e9-5dcd463c2578.png">

Reply:
@andrewkroh Nice diagram!

v2 Elasticsearch templates support components and one of the reasons we pushed for this is exactly the above problem. So long term, I'm pretty sure we will support this. Short term, I think we should get this into the development process. The current plan is to use the integrations repository for the development of the packages. There you could even use sysmlinks or whatever reference you need to in the end build the final package with all the files you need for the above. This keeps the final package simple for now.

I was not aware that netflow has 400+ fields (sounds like a lot) so in this case, the template will be pretty big in any case. Good news it is only the netflow template and does not affect anything else. Also it shouldn't be a problem if we have too many fields right now and later optimise as during update, the template will be overwritten.

Reply:
Yeah, I think some symlinks could accomplish this. Does the current module more allow for more than one fields file?

Reply:
you can have as many as you want as long as they end in `.yml` and are in the correct directory.